security/advancedtls/advancedtls.go

```diff
@@ -264,14 +274,24 @@ type ServerOptions struct {
 	// It could be nil if such checks are not needed.
 	RevocationConfig *RevocationConfig
 	// MinVersion contains the minimum TLS version that is acceptable.
-	// By default, TLS 1.2 is currently used as the minimum when acting as a
-	// client, and TLS 1.0 when acting as a server. TLS 1.0 is the minimum
-	// supported by this package, both as a client and as a server.
+	//
+	// Deprecated: use MinTLSVersion instead.
 	MinVersion uint16
 	// MaxVersion contains the maximum TLS version that is acceptable.
-	// By default, the maximum version supported by this package is used,
-	// which is currently TLS 1.3.
+	//
+	// Deprecated: use MinTLSVersion instead.
 	MaxVersion uint16
+	// MinTLSVersion contains the minimum TLS version that is acceptable.
+	// By default, TLS 1.2 is currently used as the minimum when acting as a
+	// client, and TLS 1.0 when acting as a server. TLS 1.0 is the minimum
+	// supported by this package, both as a client and as a server.  This
+	// default may be changed over time affecting backwards compatibility.
+	MinTLSVersion uint16
```

Comment:
It's really unfortunate to have to repeat all this documentation.  It seems like 99% of client and server options are identical and the only field unique to servers is `RequireClientCert`, is that right?  Can that somehow be implemented by some other feature instead, like VerifyPeer or VerificationType or something?

Or maybe we can merge them anyway and just document that field as ignored on clients.

Reply:
I agree with you - I'm very open to doing that in another PR, does that sound good to you? We can discuss specifics offline to not pollute this PR?

Reply:
SG; I wasn't trying to block this PR, just making an observation.